src/tests/tribol_enzyme_poly_intersect.cpp

```diff
@@ -0,0 +1,308 @@
+                             type, edge1, edge2 );
+      for ( int i{ 0 }; i < 16; ++i ) {
+        dxidx2_fd[16 * j + i] = x_sgn2[4 * stencil_dir[j] + j] * ( xi[i] - xi_base[i] ) / delta_;
+      }
+      x2[j] -= x_sgn2[4 * stencil_dir[j] + j] * delta_;
+
+      x2[j + 4] += y_sgn2[4 * stencil_dir[j] + j] * delta_;
+      for ( int i{ 0 }; i < 16; ++i ) {
+        xi[i] = 0.0;
+      }
+      Intersection2DPolygon( x1, x1 + 4, 4, x2, x2 + 4, 4, pos_tol, len_tol, xi, xi + 8, num_poly_verts, area, true,
+                             type, edge1, edge2 );
+      for ( int i{ 0 }; i < 16; ++i ) {
+        dxidx2_fd[16 * ( 4 + j ) + i] = y_sgn2[4 * stencil_dir[j] + j] * ( xi[i] - xi_base[i] ) / delta_;
+      }
+      x2[j + 4] -= y_sgn2[4 * stencil_dir[j] + j] * delta_;
+    }
+
+    // write deltas to screen
+    std::cout << "dxi/dx1 ------------------------------" << std::endl;
+    for ( int j{ 0 }; j < 8; ++j ) {
+      for ( int i{ 0 }; i < 16; ++i ) {
+        auto idx = j * 16 + i;
+        auto diff = std::abs( dxidx1[idx] - dxidx1_fd[idx] );
+        if ( diff > 10.0 * delta_ ) {
+          std::cout << "  (" << i << ", " << j << ") : Diff: " << diff << "   Ratio: " << dxidx1[idx] / dxidx1_fd[idx]
+                    << "   Enzyme: " << dxidx1[idx] << "   FD: " << dxidx1_fd[idx] << std::endl;
+        }
+        EXPECT_NEAR( dxidx1[idx], dxidx1_fd[idx], fd_tol );
+      }
+    }
+    std::cout << "dxi/dx2 ------------------------------" << std::endl;
+    for ( int j{ 0 }; j < 8; ++j ) {
+      for ( int i{ 0 }; i < 16; ++i ) {
+        auto idx = j * 16 + i;
+        auto diff = std::abs( dxidx2[idx] - dxidx2_fd[idx] );
+        if ( diff > 10.0 * delta_ ) {
+          std::cout << "  (" << i << ", " << j << ") : Diff: " << diff << "   Ratio: " << dxidx2[idx] / dxidx2_fd[idx]
+                    << "   Enzyme: " << dxidx2[idx] << "   FD: " << dxidx2_fd[idx] << std::endl;
+        }
+        EXPECT_NEAR( dxidx2[idx], dxidx2_fd[idx], fd_tol );
+      }
+    }
+  }
+};
+
+TEST_F( EnzymePolyIntersectTest, PerfectOverlap )
+{
+  constexpr auto pos_tol = 10.0 * delta_;
+  constexpr auto len_tol = 10.0 * delta_;
+  RealT x1[8] = { 0.0, 1.0, 1.0, 0.0, 0.0, 0.0, 1.0, 1.0 };
+  RealT x2[8] = { 0.0, 1.0, 1.0, 0.0, 0.0, 0.0, 1.0, 1.0 };
+  int stencil_dir[4] = { 0, 0, 0, 0 };
+  CheckIntersection( x1, x2, stencil_dir, delta_, pos_tol, len_tol, "perfect_overlap" );
+}
+
+TEST_F( EnzymePolyIntersectTest, Mesh2VertexMovedInByPosTol )
+{
+  constexpr auto pos_tol = 10.0 * delta_;
+  constexpr auto len_tol = 10.0 * delta_;
+  RealT x1[8] = { 0.0, 1.0, 1.0, 0.0, 0.0, 0.0, 1.0, 1.0 };
+  RealT x2[8] = { 0.0 + pos_tol, 1.0, 1.0, 0.0, 0.0, 0.0, 1.0, 1.0 };
+  int stencil_dir[4] = { 1, 1, 1, 1 };
+  CheckIntersection( x1, x2, stencil_dir, delta_, pos_tol, len_tol, "onevertpostol" );
+}
+
+// NOTE: edges are too nearly parallel which makes the derivative explode for some terms.  this makes the FD inaccurate.
```

Comment:
Perhaps clarify what this means. Do the enzyme derivatives explode? Is this an edge case that will always pose difficulties for an implicit method?

Reply:
The derivative is large since the edges are nearly parallel.  A small change in the perpendicular component of a coordinate on the edge leads to a large (linearized) change in the overlap coordinate.  Finite differencing loses precision because of this.  Added this explanation to the comment.